pkg/container-collection/options.go

```diff
@@ -585,6 +582,65 @@ func WithLinuxNamespaceEnrichment() ContainerCollectionOption {
 	}
 }
 
+// WithOCIConfigEnrichment enriches container using provided OCI config
+func WithOCIConfigEnrichment() ContainerCollectionOption {
+	return func(cc *ContainerCollection) error {
+		cc.containerEnrichers = append(cc.containerEnrichers, func(container *Container) bool {
+			if container.OciConfig == nil || container.IsEnriched() {
+				return true
+			}
+
+			// enrich crio containers
+			if cm := container.OciConfig.Annotations[crioContainerManagerAnnotation]; cm == runtimeclient.CrioName {
+				// TODO: handle this once we support pod sandboxes via WithContainerRuntimeEnrichment
+				// Issue: https://github.com/inspektor-gadget/inspektor-gadget/issues/1095
+				if ct := container.OciConfig.Annotations[crioContainerTypeAnnotation]; ct == containerTypeSandbox {
+					return false
+				}
+
+				if cn, ok := container.OciConfig.Annotations[crioContainerNameAnnotation]; ok {
+					container.Name = cn
+				}
+				if pn, ok := container.OciConfig.Annotations[crioPodNameAnnotation]; ok {
+					container.Podname = pn
+				}
+				if ns, ok := container.OciConfig.Annotations[crioPodNamespaceAnnotation]; ok {
+					container.Namespace = ns
+				}
+				if pUID, ok := container.OciConfig.Annotations[crioPodUIDAnnotation]; ok {
+					container.PodUID = pUID
+				}
+				container.Runtime = runtimeclient.CrioName
+				return true
+			}
+
+			// enrich containerd container
+			if _, isContainerd := container.OciConfig.Annotations[containerdContainerTypeAnnotation]; isContainerd {
+				// TODO: handle this once we support pod sandboxes via WithContainerRuntimeEnrichment
+				// Issue: https://github.com/inspektor-gadget/inspektor-gadget/issues/1095
+				if ct := container.OciConfig.Annotations[containerdContainerTypeAnnotation]; ct == containerTypeSandbox {
+					return false
+				}
+
+				if cn, ok := container.OciConfig.Annotations[containerdContainerNameAnnotation]; ok {
+					container.Name = cn
+				}
+				if pn, ok := container.OciConfig.Annotations[containerdPodNameAnnotation]; ok {
+					container.Podname = pn
+				}
+				if ns, ok := container.OciConfig.Annotations[containerdPodNamespaceAnnotation]; ok {
+					container.Namespace = ns
+				}
+				container.Runtime = runtimeclient.ContainerdName
+				return true
+			}
+
+			return true
```

Comment:
I thought something like this could look better, but I am not that sure anymore:

```suggestion
			var containerTypeAnnotation, containerNameAnnotation, podNameAnnotation, podNamespaceAnnotation string

			if cm := container.OciConfig.Annotations[crioContainerManagerAnnotation]; cm == runtimeclient.CrioName {
				container.Runtime = runtimeclient.CrioName

				containerTypeAnnotation = crioContainerTypeAnnotation
				containerNameAnnotation = crioContainerNameAnnotation
				podNameAnnotation = crioPodNameAnnotation
				podNamespaceAnnotation = crioPodNamespaceAnnotation

				if pUID, ok := container.OciConfig.Annotations[crioPodUIDAnnotation]; ok {
					container.PodUID = pUID
				}
			} else if _, isContainerd := container.OciConfig.Annotations[containerdContainerTypeAnnotation]; isContainerd {
				container.Runtime = runtimeclient.ContainerdName

				containerTypeAnnotation = containerdContainerTypeAnnotation
				containerNameAnnotation = containerdContainerNameAnnotation
				podNameAnnotation = containerdPodNameAnnotation
				podNamespaceAnnotation = containerdPodNamespaceAnnotation
			} else {
				return true
			}

			// TODO: handle this once we support pod sandboxes via WithContainerRuntimeEnrichment
			// Issue: https://github.com/inspektor-gadget/inspektor-gadget/issues/1095
			if ct := container.OciConfig.Annotations[containerTypeAnnotation]; ct == containerTypeSandbox {
				return false
			}

			if cn, ok := container.OciConfig.Annotations[containerNameAnnotation]; ok {
				container.Name = cn
			}
			if pn, ok := container.OciConfig.Annotations[podNameAnnotation]; ok {
				container.Podname = pn
			}
			if ns, ok := container.OciConfig.Annotations[podNamespaceAnnotation]; ok {
				container.Namespace = ns
			}

			return true
```